### ISSUE TITLE
Ordinal axes and linkedTo

### ISSUE THREAD

**Issue by pawelfus** [highslide-software/highcharts.com#1417]
Example:  http://jsfiddle.net/vp6m4/24/

As we can see, labels are not properly aligned.

**Comment by maximenicot74** [comment]
This issue still not being solved nowadays.

I want to plot flags an a second axis, but flags are not aligned with column or candlesticks, or whatever.
http://jsfiddle.net/bgt6omev/

Could we know if this will be fixed soon, or do I have to find an alternative?

**Comment by ItaloFSS** [comment]
I found a solution that works for what I need. What I need is to clone the primary x-axis, in every way possible, from the number of ticks to their positions; having different x-axis confuses the user. So, what I did is override the axis prototype to always use the primary x-axis ticks, and translate function. This worked great, I now have two identical x-axes, and the labels aligns perfectly.

Here's the code:

var originalSetTickPositions = Highcharts.Axis.prototype.setTickPositions;
Highcharts.Axis.prototype.setTickPositions = function()
{
if( this.horiz && this.linkedParent && this.linkedParent.tickPositions )
{
this.tickmarkOffset = this.linkedParent.tickmarkOffset;
this.minorTickInterval = this.linkedParent.minorTickInterval;
this.tickPositions = this.linkedParent.tickPositions;
return;
}

originalSetTickPositions.call( this );
}

var originalTranslate = Highcharts.Axis.prototype.translate;
Highcharts.Axis.prototype.translate = function( val, backwards, cvsCoord, old, handleLog, pointPlacement )
{
if( this.horiz && this.linkedParent )
{
return this.linkedParent.translate( val, backwards, cvsCoord, old, handleLog, pointPlacement );
}

return originalTranslate.call( this, val, backwards, cvsCoord, old, handleLog, pointPlacement );
}
